### PR TITLE
Add player fish kinematic physics 

### LINKED ISSUE #20: Make Player a Fish
This includes making them subject to
--> water physics and other Kinematic physics interactions
--> fish animation
--> being eaten (food target) 
-->

### PR DESCRIPTION
**Pull request by Akisukei**

### CODE REVIEW

**Comment by Akisukei** [comment]
I also realized I didn't implement the collision detection (been using RigidBody on player) so I'll have to add that in too

**Review by Taqi — APPROVED** [review]
DAM SON, the gameplay looks great!

File: Assets/Scripts/Fish/Fish.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+    private void Awake() => kinematic = new Kinematic();
+
+    void Update()
+    {
+        HandlePhysics();
+
+        // Calculate transform with kinematic linear movement + any other force applied
+        transform.position = transform.position + (kinematic.LinearVel * Time.deltaTime) + (momentum * Time.deltaTime);
+        transform.rotation = kinematic.Orientation;
```
Comment (jmcduff) [inline Assets/Scripts/Fish/Fish.cs:25]: Currently, the Flock script takes care of applying velocities on its Flock Agents. I suppose we could do a refactoring so Flocks rely instead on the Fish script 🤔 I'll look into that once you've merged this PR.

**Review by jmcduff — APPROVED** [review]
Looks good!

**Pull request merged**